msg.go

```diff
@@ -427,10 +427,7 @@ Loop:
 		off1 = off
 	}
 	if len(s) == 0 {
-		s = []byte(".")
-	} else if len(s) >= maxLen {
-		// error if the name is too long, but don't throw it away
-		return string(s), lenmsg, ErrLongDomain
```

Comment:
maybe update the doc comment about this behavior?

Reply:
@miekg Would you prefer a comment on `UnpackDomainName` or where the `if budget <= 0` return is?

Reply:
How is that?